### DISCUSSION TITLE
Docker: Tasmota doesn't cleanly compile since 13.0

### DISCUSSION THREAD

**Discussion by blacknell** [arendst/Tasmota#19086] [Support request]
I've used docker-tasmota to compile and build Tasmota for several releases.

Since 13.0 the compile is failing with the following error output. Weirdly, the firmware binaries are still created in the expected output directory.

Here I'm building `release` with some custom defines - happy to post fuller detail if the error message doesn't point to an obvious cause.

```
Wrote 0x2336f0 bytes to file /tasmota/.pio/build/tasmota32-thermostat/firmware.factory.bin, ready to flash to offset 0x0
bin_map_copy([".pio/build/tasmota32-thermostat/firmware.bin"], [".pio/build/tasmota32-thermostat/firmware.elf"])
map_gzip([".pio/build/tasmota32-thermostat/firmware.bin"], [".pio/build/tasmota32-thermostat/firmware.elf"])
firm_metrics([".pio/build/tasmota32-thermostat/firmware.bin"], [".pio/build/tasmota32-thermostat/firmware.elf"])
*** [.pio/build/tasmota32-thermostat/firmware.bin] ModuleNotFoundError : No module named 'tasmota_metrics'
Traceback (most recent call last):
  File "/.platformio/packages/tool-scons/scons-local-4.5.2/SCons/Action.py", line 1310, in execute
    result = self.execfunction(target=target, source=rsources, env=env)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/tasmota/pio-tools/metrics-firmware.py", line 9, in firm_metrics
    import tasmota_metrics
ModuleNotFoundError: No module named 'tasmota_metrics'
============================================================ [FAILED] Took 603.56 seconds ============================================================

```

**Comment by Jason2866** [comment]
It is not a Tasmota issue. The built Docker image does not install the needed tasmota-metrics correctly.
Just a minor issue since tasmota-metrics just shows the needed resources of the the firmware.
The firmware(s) will be built correctly and can be used!

The Docker setup is here https://github.com/tasmota/docker-tasmota which needs fixing for installing tasmota-metrics from pypi and make it useable for Platformio. 
There is an open PR. But the way done is not optimal. https://github.com/tasmota/docker-tasmota/pull/22

**Reply by barbudor** [reply]
@Jason2866 I have comment back to that PR
For me there is no other way of doing it

**Reply by Jason2866** [reply]
@barbudor After doing some (more) research, i agree your PR is the way to go.

**Comment by blacknell** [comment]
FYI. I made the changes manually in my docker-tasmota and build completed without errors. I expect this is not a surprise!
